packages/network-enablement-controller/src/NetworkEnablementController.ts

```diff
@@ -0,0 +1,482 @@
+import { BaseController } from '@metamask/base-controller';
+import type {
+  ControllerGetStateAction,
+  ControllerStateChangeEvent,
+  RestrictedMessenger,
+} from '@metamask/base-controller';
+import { BuiltInNetworkName, ChainId, toHex } from '@metamask/controller-utils';
+import { SolScope } from '@metamask/keyring-api';
+import type { MultichainNetworkControllerGetStateAction } from '@metamask/multichain-network-controller';
+import { toEvmCaipChainId } from '@metamask/multichain-network-controller';
+import type {
+  NetworkControllerGetStateAction,
+  NetworkControllerNetworkAddedEvent,
+  NetworkControllerNetworkRemovedEvent,
+  NetworkControllerStateChangeEvent,
+} from '@metamask/network-controller';
+import type { CaipChainId, CaipNamespace, Hex } from '@metamask/utils';
+import {
+  isCaipChainId,
+  isHexString,
+  KnownCaipNamespace,
+  parseCaipChainId,
+} from '@metamask/utils';
+
+import { POPULAR_NETWORKS } from './constants';
+import { selectAllEnabledNetworks } from './selectors';
+
+// Unique name for the controller
+const controllerName = 'NetworkEnablementController';
+
+/**
+ * Information about an ordered network.
+ */
+export type NetworksInfo = {
+  /**
+   * The network's chain id
+   */
+  networkId: CaipChainId;
+};
+
+/**
+ * A map of enabled networks by namespace and chain id.
+ */
+type EnabledMap = Record<CaipNamespace, Record<string, boolean>>;
```

Comment:
Because the `CaipNamespace` can be deleted, can we make the value optional? Also thoughts on expanding this type so we can insert some JSDoc comments?

```ts
interface EnabledMap {
    /**
     * caipNamespace
     * @example eip155 or solana
     * NOTE - a namespace may be deleted, hence indexing can result in undefined.
     */
    [caipNamespace: string]: {
        /**
         * chainId, provided as hex or CAIP-2
         * @example 0x1 or solana:mainnet
         */
        [chainId: string]: boolean
    } | undefined
}
```